src/common/snippets/src/pass/split_dimension_m.cpp

```diff
@@ -34,23 +34,23 @@ bool SplitDimensionM::is_supported_matmul(const std::shared_ptr<const ov::Node>&
 std::pair<size_t, size_t> SplitDimensionM::get_splited_dimensions(size_t batch_dim, size_t m_dim, size_t optimal_parallelism_work_amount) {
     std::pair<size_t, size_t> splited = { 1, m_dim };
 
-    const size_t lower_bound = optimal_parallelism_work_amount / batch_dim;
-    if (lower_bound * batch_dim == optimal_parallelism_work_amount && m_dim % lower_bound == 0) {
-        splited.first = lower_bound;
-        splited.second = m_dim / lower_bound;
-        OPENVINO_ASSERT(splited.first * splited.second == m_dim, "Incorrect dimension M splitting!");
-        return splited;
-    }
-
-    const size_t upper_bound = utils::div_up(2 * optimal_parallelism_work_amount, batch_dim);
-    for (size_t divisor_0 = upper_bound - 1; divisor_0 > 1; divisor_0--) {
-        size_t divisor_1 = m_dim / divisor_0;
-        if (divisor_1 * divisor_0 == m_dim) {
-            splited.first = divisor_0;
-            splited.second = divisor_1;
-            break;
+    // TODO: should we limit minimal kernel_m?
+    const size_t min_kernel_m = 4;
```

Comment:
I'm not sure, but I believe than `min_kernel_m` should be 32-64 at least. For me, `4` is too low value. However, let's wait for the perf validation 😃